pkg/controller/servicebindingrequest/binder.go

```diff
@@ -0,0 +1,295 @@
+	case "deployment":
+		return appsv1.Resource("deployments").WithVersion("v1"), nil
+	default:
+		return schema.GroupVersionResource{},
+			fmt.Errorf("resource kind '%s' is not supported by this operator", kind)
+	}
+}
+
+// search objects based in Kind/APIVersion, which contain the labels defined in ApplicationSelector.
+func (b *Binder) search() (*ustrv1.UnstructuredList, error) {
+	gvr, err := b.getListGVR()
+	if err != nil {
+		return nil, err
+	}
+
+	ns := b.sbr.GetNamespace()
+	matchLabels := b.sbr.Spec.ApplicationSelector.MatchLabels
+	opts := metav1.ListOptions{
+		LabelSelector: labels.Set(matchLabels).String(),
+	}
+
+	return b.dynClient.Resource(gvr).Namespace(ns).List(opts)
+}
+
+// updateSpecVolumes execute the inspection and update "volumes" entries in informed spec.
+func (b *Binder) updateSpecVolumes(
+	logger logr.Logger,
+	obj *ustrv1.Unstructured,
+) (*ustrv1.Unstructured, error) {
+	volumesPath := []string{"spec", "template", "spec", "volumes"}
+	logger = logger.WithValues("Volumes.NestedPath", volumesPath)
+
+	logger.Info("Reading volumes definitions...")
+	volumes, _, err := ustrv1.NestedSlice(obj.Object, volumesPath...)
+	if err != nil {
+		return nil, err
+	}
+	logger.WithValues("Volumes", len(volumes)).Info("Amount of volumes in spec.")
+
+	volumes, err = b.updateVolumes(logger, volumes)
+	if err != nil {
+		return nil, err
+	}
+	if err = ustrv1.SetNestedSlice(obj.Object, volumes, volumesPath...); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+// updateVolumes inspect informed list assuming as []corev1.Volume, and if binding volume is already
+// defined just return the same list, otherwise, appending the binding volume.
+func (b *Binder) updateVolumes(logger logr.Logger, volumes []interface{}) ([]interface{}, error) {
+	name := b.sbr.GetName()
+
+	logger.Info("Checking if binding volume is already defined...")
+	for _, v := range volumes {
+		volume := v.(corev1.Volume)
+		if name == volume.Name {
+			logger.Info("Volume is already defined!")
+			return volumes, nil
+		}
+	}
+
+	items := []corev1.KeyToPath{}
+	for _, k := range b.volumeKeys {
+		items = append(items, corev1.KeyToPath{Key: k, Path: k})
+	}
+
+	logger.WithValues("Items", items).Info("Appending new volume with items.")
+	bindVolume := corev1.Volume{
+		Name: name,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: name,
+				Items:      items,
+			},
+		},
+	}
+
+	// making sure tranforming it back to unstructured before returning
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&bindVolume)
+	if err != nil {
+		return nil, err
+	}
+	return append(volumes, u), nil
+}
+
+// updateSpecContainers extract containers from object, and trigger update.
+func (b *Binder) updateSpecContainers(
+	logger logr.Logger,
+	obj *ustrv1.Unstructured,
+) (*ustrv1.Unstructured, error) {
+	containersPath := []string{"spec", "template", "spec", "containers"}
```

Comment:
@sbose78 yep, those kind are now supported since the internal exchange format is `Unstructured` and thus we can add a simple `if` block to address resources that would not follow the same structured.